packages/server/utils/roomConfig.ts

```diff
@@ -0,0 +1,78 @@
+import Handle from './handle';
+import { ClientId } from '../objects/client';
+import {
+    getFailure,
+    getJoinSuccess,
+    getCreateRoomSuccess,
+    getLeaveSuccess,
+    getGameStartSuccess,
+} from '../message';
+import { MessageType } from '@prisel/common';
+import { GAME_PHASE } from '../objects/gamePhase';
+
+type EventHandler = (handle: Handle, client: ClientId, data: any) => void;
+
+export interface RoomConfig {
+    type: string;
+    onCreate: EventHandler;
+    onJoin: EventHandler;
+    onLeave: EventHandler;
+    onGameStart: EventHandler;
+    onMessage: EventHandler;
+}
+
+const addClientToRoom = (handle: Handle, client: ClientId) => {
+    if (handle.gamePhase === GAME_PHASE.WAITING) {
+        handle.addClient(client);
+        handle.game.addPlayer(handle, client);
```

Comment:
We are able to add a client only before the game stared?
Could we do something like:
// add the function addPlayerAfterStart() to gameConfig

handle.addClient(client);
if (handle.gamePhase === GAME_PHASE.WAITING){
    handle.game.addPlayer(handle, client);
}else if(handle.gamePhase === GAME_PHASE.GAME){
    handle.game.addPlayerAfterStart(handle, client);
} 
--------------------------------------------
Or we don't check the game phase here. When a client comes, we always call addPlayer. Users check the game phase in the addPlayer function?